src/uint.rs

```diff
@@ -63,21 +63,30 @@ macro_rules! uint_overflowing_add {
 
 macro_rules! uint_overflowing_add_reg {
 	($name:ident, $n_words:expr, $self_expr: expr, $other: expr) => ({
-		let $name(ref me) = $self_expr;
-		let $name(ref you) = $other;
-
-		let mut ret = [0u64; $n_words];
-		let mut carry = 0u64;
-
-		for i in 0..$n_words {
-			let (res1, overflow1) = me[i].overflowing_add(you[i]);
-			let (res2, overflow2) = res1.overflowing_add(carry);
+		unsafe {
+			let mut carry = 0u64;
+			let mut ret: [u64; $n_words] = mem::uninitialized();
+			let mut me_ptr = $self_expr.0.as_ptr();
+			let mut you_ptr = $other.0.as_ptr();
+			let mut ret_ptr = ret.as_mut_ptr();
+			for _ in 0..$n_words {
+				let (res1, overflow1) = (*me_ptr).overflowing_add(*you_ptr);
+
+				if carry != 0 {
```

Comment:
@NikVolf did you disable assembly for nightly? also, together with @Vurich we found a way to parallelize the operations, so the addition and substraction is almost 10x faster, and even faster than assembly code. For now, I'm closing the pr, and we will open the new one tomorrow